### PR TITLE
Introduce buttons to create users, groups and roles from the respective provider

### PR DESCRIPTION
**Pull request by dnlkoch**
This adds a button to each of the `User`, `Group` and `Role` panels to trigger the creation of the given entities from the respective provider (see [#947](https://github.com/terrestris/shogun/pull/947)). Generally it just adds all missing entities, e.g. all users that are present in the user provider (usually Keycloak), but not in SHOGun.

Peek 2024-11-11 13-18

In addition this:

- Refactors the `GeneralEntityRoot` for more flexible and generic support for entity specific components to render (currently only for the toolbar above the entity table, but this might be adjusted for future needs) by introducing a local context including some props that were shared between all sub components.
- Fixes a regression of missing support for additional navigation menu entries besides the default ones.

Requires [#821](https://github.com/terrestris/shogun-util/pull/821).

Please review @terrestris/devs.

### CODE REVIEW

File: src/Component/CreateAllGroupsButton/CreateAllGroupsButton.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+jest.mock('../../Hooks/useSHOGunAPIClient', () => {
+  const originalModule = jest.requireActual('../../Hooks/useSHOGunAPIClient');
+  return {
+    __esModule: true,
+    ...originalModule,
+    default: jest.fn(() => mockSHOGunAPIClient)
+  };
+});
+
+describe('<CreateAllUsersButton />', () => {
```
Comment (LukasLohoff) [inline src/Component/CreateAllGroupsButton/CreateAllGroupsButton.spec.tsx:36]:
```suggestion
describe('<CreateAllGroupsButton />', () => {
```

File: src/i18n/translations.ts [hunk 2]
```diff
@@ -250,10 +236,42 @@ export default {
...
+      },
+      CreateAllUsersButton: {
+        title: 'Nutzer synchronisieren',
+        tooltip: 'Add all missing users from the user provider',
+        success: 'Nutzer erfolgreich erstellt',
+        error: 'Fehler beim Erstellen der Nutzer'
+      },
+      CreateAllGroupsButton: {
+        title: 'Gruppen synchronisieren',
+        tooltip: 'Add all missing users from the user provider',
```
Comment (LukasLohoff) [inline src/i18n/translations.ts:265]:
```suggestion
        tooltip: 'Add all missing groups from the group provider',
```

File: src/i18n/translations.ts [hunk 3]
```diff
@@ -250,10 +236,42 @@ export default {
...
+        success: 'Gruppen erfolgreich erstellt',
+        error: 'Fehler beim Erstellen der Gruppen'
+      },
+      CreateAllRolesButton: {
+        title: 'Rollen synchronisieren',
+        tooltip: 'Add all missing users from the user provider',
```
Comment (LukasLohoff) [inline src/i18n/translations.ts:271]:
```suggestion
        tooltip: 'Add all missing roles from the role provider',
```

Comment (LukasLohoff) [reply src/i18n/translations.ts:271]: by the way, shouldn't this be in German?

**Review by LukasLohoff — APPROVED** [review]

File: src/Component/CreateAllGroupsButton/CreateAllGroupsButton.spec.tsx (hunk 1 above)
Comment (dnlkoch) [reply src/Component/CreateAllGroupsButton/CreateAllGroupsButton.spec.tsx]: Fixed in 38d4309ef04c93a5bd9dfc5fc663e76cfa8a3399.

File: src/i18n/translations.ts (hunk 3 above)
Comment (dnlkoch) [reply src/i18n/translations.ts]: Thanks, nice catch, you're absolutely right!

Fixed in 38d4309ef04c93a5bd9dfc5fc663e76cfa8a3399.

File: src/i18n/translations.ts (hunk 2 above)
Comment (dnlkoch) [reply src/i18n/translations.ts]: Fixed in 38d4309ef04c93a5bd9dfc5fc663e76cfa8a3399.

**Comment by dnlkoch** [comment]
Thanks for the review @LukasLohoff!

**Pull request merged**